### PR TITLE
Add warning for mixed version range syntax

### PR DESCRIPTION
**Pull request by domcleal**
semantic_puppet 1.0.0 now accepts a mixed range syntax, per the NodeJS
specifications. It should be considered as bad style though as the
syntax is unclear and will only be available in Puppet 5 and higher.

---

Fixes the test failure of mixed_version_syntax on master, which occurred after the release of semantic_puppet 1.0.0 by changing it to expect the new warning.

See https://groups.google.com/forum/#!topic/puppet-dev/BohRo-fqy-g for discussion about whether this is valid or not.

### CODE REVIEW

File: spec/version_requirement_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+describe MetadataJsonLint::VersionRequirement do
+  describe '.new' do
+    it { expect(described_class.new('')).to be_a(MetadataJsonLint::VersionRequirement) }
+    it { expect(described_class.new('>= 1.0')).to be_a(MetadataJsonLint::VersionRequirement) }
+    it { expect { described_class.new('## 1.0') }.to raise_error(ArgumentError) }
+  end
+
+  describe '#mixed_syntax?' do
+    it { expect(described_class.new('>= 1.0.0').mixed_syntax?).to be false }
+    it { expect(described_class.new('1.0.x').mixed_syntax?).to be false }
```
Comment (rnelson0) [inline spec/version_requirement_spec.rb]: Why isn't this considered mixed syntax?

Comment (domcleal) [inline spec/version_requirement_spec.rb]: It only uses `.x` - if it was mixed syntax then it'd also be using `>=` or similar.

`.x` by itself is reasonably clear, i.e. 1.0.0 through to 1.0.999, but adding a second operator such as `>= 1.0.x` is what it tries to warn against, as the intent is less clear.

Comment (rnelson0) [inline spec/version_requirement_spec.rb]: Hrm, then why does the mixed_version_syntax test produce a warning with `3.x` without any second operator?

Comment (domcleal) [inline spec/version_requirement_spec.rb]: The mixed_version_syntax test has `>= 3.2.x` in metadata.json for the dependency, I think you're looking at the `puppet` requirement which doesn't have the same check.

Comment (rnelson0) [inline spec/version_requirement_spec.rb]: Ah, that makes more sense.

**Pull request merged**